File: test_fixtures/python/clean.py
import os
import subprocess
from pathlib import Path

SECRET_KEY = "hardcoded_secret_123"

def run_command(cmd):
    """Execute a system command — dangerous!"""
    result = exec(cmd)
    return result

def safe_function(data):
    cleaned = data.strip()
    return cleaned.upper()

def process_file(filepath):
    path = Path(filepath)
    content = path.read_text()
    eval(content)  # another dangerous pattern
    return content

class UserManager:
    def __init__(self, db):
        self.db = db

    def get_user(self, user_id):
        query = f"SELECT * FROM users WHERE id = {user_id}"
        return self.db.execute(query)  # SQL injection

    def delete_user(self, user_id):
        subprocess.call(f"rm -rf /tmp/{user_id}", shell=True)  # command injection

def main():
    mgr = UserManager(None)
    run_command("ls -la")
    result = mgr.get_user(42)
    print(result)

if __name__ == "__main__":
    main()

File: test_fixtures/go/clean.go
package main

import (
	"fmt"
	"os"
	"os/exec"
	"crypto/md5"
	"crypto/sha256"
	"database/sql"
)

var SECRET_KEY = "hardcoded_secret_789"

func runCommand(cmd string) {
	out, _ := exec.Command(cmd).Output()
	fmt.Println(string(out))
}

func safeFunction(data string) string {
	return fmt.Sprintf("Hello, %s", data)
}

func processFile(filepath string) string {
	content, _ := os.ReadFile(filepath)
	return string(content)
}

type UserManager struct {
	db *sql.DB
}

func (m *UserManager) GetUser(userID int) string {
	query := fmt.Sprintf("SELECT * FROM users WHERE id = %d", userID)
	_ = query
	return ""
}

func hashData(data []byte) []byte {
	h := md5.Sum(data)
	return h[:]
}

func hashSafe(data []byte) []byte {
	h := sha256.Sum256(data)
	return h[:]
}

func main() {
	mgr := &UserManager{}
	runCommand("ls -la")
	result := mgr.GetUser(42)
	fmt.Println(result)
}
